src/libraries/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -530,7 +530,7 @@ private static void SetCurrentContextualReflectionContext(AssemblyLoadContext? v
             {
                 Interlocked.CompareExchange<AsyncLocal<AssemblyLoadContext?>?>(ref s_asyncLocalCurrent, new AsyncLocal<AssemblyLoadContext?>(), null);
             }
-            s_asyncLocalCurrent!.Value = value; // Remove ! when compiler specially-recognizes CompareExchange for nullability
+            s_asyncLocalCurrent.Value = value; // Remove ! when compiler specially-recognizes CompareExchange for nullability
```

Comment:
```suggestion
            s_asyncLocalCurrent.Value = value;
```

Reply:
Will make this change once `main` is open for .NET 11 changes.

Reply:
main is open for .NET 11 changes.